go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+			inputLink:   mockLink,
+			outBoolFlag: false,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"NeighList", []string{"int", "int"}, []interface{}{[]netlink.Neigh{}, nil}},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName", Index: 1}}},
+			},
+		},
+		{
+			desc:         "test path where MAC/IP binding is established",
+			inputLink:    mockLink,
+			inputNeigIP:  ovntest.MustParseIP("192.169.1.12"),
+			inputMacAddr: ovntest.MustParseMAC("0A:58:FD:98:00:01"),
+			outBoolFlag:  true,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"NeighList", []string{"int", "int"},
+					[]interface{}{
+						[]netlink.Neigh{
+							{IP: ovntest.MustParseIP("192.169.1.12"), HardwareAddr: ovntest.MustParseMAC("0A:58:FD:98:00:01"), State: netlink.NUD_PERMANENT},
+						},
+						nil,
+					},
+				},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName", Index: 1}}},
+			},
+		},
+		{
+			desc:         "test path where MAC/IP bindings DOES NOT exist",
+			inputLink:    mockLink,
+			inputNeigIP:  ovntest.MustParseIP("192.169.1.15"),
+			inputMacAddr: ovntest.MustParseMAC("0A:58:FD:98:00:01"),
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"NeighList", []string{"int", "int"},
+					[]interface{}{
+						[]netlink.Neigh{
+							{IP: ovntest.MustParseIP("192.169.1.12"), HardwareAddr: ovntest.MustParseMAC("0A:58:FD:98:00:01"), State: netlink.NUD_PERMANENT},
+						},
+						nil,
+					},
+				},
+			},
+			onRetArgsLinkIfaceOpers: []onCallReturnArgs{
+				{"Attrs", []string{}, []interface{}{&netlink.LinkAttrs{Name: "testIfaceName", Index: 1}}},
+			},
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			flag, err := LinkNeighExists(tc.inputLink, tc.inputNeigIP, tc.inputMacAddr)
+			t.Log(flag, err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
+			if tc.outBoolFlag {
+				assert.True(t, flag)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+			mockLink.AssertExpectations(t)
+		})
+	}
+}
+
+func TestDeleteConntrack(t *testing.T) {
```

Comment:
Should the test names start with capitol letter? I think you can make it package scoped as we are not exporting anything unless I am missing any semantics

Reply:
func TestXxx(*testing.T) is the signature structure per https://golang.org/pkg/testing/